core/src/services/redis/backend.rs

```diff
@@ -185,7 +185,18 @@ impl Builder for RedisBuilder {
                 let port = ep_url.port_u16().unwrap_or(DEFAULT_REDIS_PORT);
                 ConnectionAddr::Tcp(host, port)
             }
-            // TODO: wait for upstream to support `rustls` based TLS connection.
+            Some("rediss") => {
+                let host = ep_url
+                    .host()
+                    .map(|h| h.to_string())
+                    .unwrap_or_else(|| "127.0.0.1".to_string());
+                let port = ep_url.port_u16().unwrap_or(DEFAULT_REDIS_PORT);
+                ConnectionAddr::TcpTls {
+                    host,
+                    port,
+                    insecure: false,
+                }
+            }
```

Comment:
Maybe we need to add the Redis TLS Integration test

Reply:
We can raise a new issue for this

Reply:
sure thing
May i ask for some lead on where to put this test ? I cannot find redis integration tests

Reply:
OpenDAL has behavior tests for all tests. You can check out our github workflows for more details.

Reply:
added a new test for TLS

Please tell me if there is anything wrong with it, I am very much a beginner with github actions.

Also, it could be interesting to move the certification creation in an action if you were to need TLS tests for other services.

Reply:
Thanks a lot!